toffy/rosetta.py

```diff
@@ -302,25 +302,45 @@ def replace_with_intensity_image(base_dir, channel, replace=True, folders=None):
 
 def create_rosetta_matrices(default_matrix, save_dir, multipliers=[0.5, 1, 1.5], channels=None):
     """Creates a series of compensation matrices for evaluating coefficients
-
     Args:
         default_matrix (str): path to the rosetta matrix to use as the default
         multipliers (list): the range of values to multiply the default matrix by
             to get new coefficients
         channels (list | None): an optional list of channels to include in the multiplication. If
             only a subset of channels are specified, other channels will retain their values
-            in all iterations. If None, all channels are included"""
-
-    # TODO: Cameron will make this function
+            in all iterations. If None, all channels are included
+    """
 
     # step 1: read in the default matrix
-
-    # step 2: figure out which channels will be modified
-
-    # step 3: loop over each of the multipliers
-
+    comp_matrix = pd.read_csv(default_matrix, index_col=0) # pandas DataFrame
+    row_labels = comp_matrix.index
+    comp_channels = list(row_labels)
+    matrix_rows = len(comp_matrix)
+    matrix_columns = len(comp_matrix.iloc[0])
+    column_features = list(comp_matrix.columns.values)
+
+    # step 2: figure out which channels will be modified (rows)
+    if channels is None:
+        channels = comp_channels
+
+    for i in channels:
+        if i not in comp_channels:
+            raise ValueError('Specified channel does not exist')
+
+    # step 3: loop over each of the multipliers (anything the user inputs, make an output matrix)
     # step 4: modify the appropriate channels based on mulitiplier
-
     # step 5: save the modified matrix as original_name_multiplier
+    for i in multipliers: # returns each comp_matrix value
+        zero_matrix = np.zeros(shape=(matrix_rows+1, matrix_columns))
+        modified_matrix = pd.DataFrame(zero_matrix[1:], index = row_labels, columns=column_features)
+        for j in range(matrix_rows):
+            for k in channels:
```

Comment:
The logic here is a bit muddled. 

You first loop over each `j` row, where each row corresponds to a channel. 
However, after you've picked the `jth` row, you then loop over each channel, and for every channel, check if it's part of comp_channels. 

You then repeat this same thing for every time you pick a new row. At the end of this, you then multiply the whole thing by zero?